src/plugins/plugin.filler.js

```diff
@@ -250,11 +291,45 @@ function doFill(ctx, points, mapper, view, color, loop) {
 		}
 	}
 
-	drawArea(ctx, curve0, curve1, len0, len1);
+	sets = drawArea(ctx, curve0, curve1, len0, len1, mode, area);
+	if (sets) {
+		clipAreaPointsSets.push(sets[0]);
+		fillAreaPointsSets.push(sets[1]);
+	}
+
+	if (mode !== 'all') {
+		for (i = 0; i < clipAreaPointsSets.length; i++) {
```

Comment:
Should bother using `ilen`, for consistency (and maybe a tiny performance boost):
```js
for (i = 0, ilen = clipAreaPointsSets.length; i< ilen; ++i) {
```